src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowHistoryV2Spec.groovy

```diff
@@ -298,6 +306,42 @@ class FlowHistoryV2Spec extends HealthCheckSpecification {
         ]
     }
 
+    @Tidy
+    @Tags(LOW_PRIORITY)
+    def "Info about a terminating switch is registered in flow history while rerouting"() {
+        given: "An active flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def flow = flowHelperV2.randomFlow(srcSwitch, dstSwitch)
+        flowHelperV2.addFlow(flow)
+
+        when: "Deactivate the src switch"
+        def blockData = lockKeeper.knockoutSwitch(srcSwitch, RW)
+        def swIsActive = false
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getSwitch(srcSwitch.dpId).state == SwitchChangeType.DEACTIVATED
+        }
+
+        then: "Flow goes DOWN"
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            def flowInfo = northboundV2.getFlow(flow.flowId)
+            assert flowInfo.status == FlowState.DOWN.toString()
+            assert flowInfo.statusInfo == "ValidateFlowAction failed: Flow's $flow.flowId src switch is not active"
+        }
+
+        and: "The 'Switch is not active' message is registered in flow history"
+        def flowHistory = northbound.getFlowHistory(flow.flowId).find { it.action == REROUTE_ACTION }
+        assert flowHistory.payload[0].action == "Started flow validation"
+        assert flowHistory.payload[1].action == "ValidateFlowAction failed: Flow's $flow.flowId src switch is not active"
+        assert flowHistory.payload[2].action == REROUTE_FAIL
+
+        cleanup:
+        if (!swIsActive) {
+            lockKeeper.reviveSwitch(srcSwitch, blockData)
+            Wrappers.wait(WAIT_OFFSET) { northbound.getSwitch(srcSwitch.dpId).state == SwitchChangeType.ACTIVATED }
```

Comment:
It's better to use `switchHelper.reviveSwitch` instead

Reply:
fixed